### DISCUSSION TITLE
crypto_secretbox_easy produces different results on 32 bit and 64 bit processes given same values

### DISCUSSION THREAD

**Discussion by randynmary** [jedisct1/libsodium#1399] [Q&A]
crypto_secretbox_easy in my testing produces different encryption results for same message, subkey, context and nonce on 32 bit and 64 bit processes.  I have been using libsodium 1.0.20 for encryption of data at rest and my tests are showing that crypto_secretbox_easy produces a different encrypted value depending on whether the x64 or x86 (32 bit) version of the libsodium.dll is picked up.  Is this expected?  It makes crypto_secretbox_easy unuseable as in future we may migrate the application from 32 to 64 bit or a 64 bit application decrypting the data will be unable to do so.  Is there a way to resolve this by calling a lower-level method or other way to make the 32 and 64bit versions produce the same results?  Thanks.

**Comment by jedisct1** [comment]
This isn't expected, and may be an issue with the bindings or your application.
